internal/service/amplify/domain_association.go

```diff
@@ -62,6 +62,7 @@ func resourceDomainAssociation() *schema.Resource {
 						names.AttrType: {
 							Type:             schema.TypeString,
 							Required:         true,
```

Comment:
This won't work unless type changed to `Optional: true`.

I fixed the same issue while working on another bug in my PR:
- https://github.com/hashicorp/terraform-provider-aws/pull/40351